astropy/io/fits/tests/test_header.py

```diff
@@ -603,9 +603,7 @@ def test_hierarch_not_warn(self):
         filename = fits.util.get_testdata_filepath('compressed_image.fits')
         with fits.open(filename) as hdul:
             header = hdul[1].header
-        with pytest.warns(None) as warning_list:
-            header["HIERARCH LONG KEYWORD"] = 42
-        assert len(warning_list) == 0
```

Comment:
Unless this is emitting a warning that we are purposely ignoring in `setup.cfg`, we do not need a check.

Reply:
This probably comes from before the switch to use pytest's warning system.  But given that the aim of this test is to test that no warning is emitted, it probable makes sense to check that explicitly ? Or at least add a comment explaining that if a warning is emitted this would fail the test.

Reply:
Okay, I added them back.